src/select/internal.tsx

```diff
@@ -272,7 +274,7 @@ const InternalSelect = React.forwardRef(
             highlightType={highlightType}
           />
         </Dropdown>
-        <ScreenreaderOnly id={selectAriaLabelId}>{ariaLabel}</ScreenreaderOnly>
+        <ScreenreaderOnly id={selectAriaLabelId}>{ariaLabel || inlineLabelText}</ScreenreaderOnly>
```

Comment:
could this be skipped if we have `<label for={controlId}>`?

Reply:
The trigger button gets an aria-labelledby that would override the `for` association so I used this node because its referenced by that aria-labelledby